subprojects/core/src/integTest/groovy/org/gradle/api/file/FileContentsIntegrationTest.groovy

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.file
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import spock.lang.Issue
+
+import java.nio.charset.Charset
+
+@Issue("https://github.com/gradle/gradle/issues/15802")
+class FileContentsIntegrationTest extends AbstractIntegrationSpec  {
+
+    static final String TURKISH_TEXT = "Ğğ, İ, ı, Şş"
+
+    def "succeeds with Turkish encoding"() {
+        given:
+        testDirectory.file("foo.txt").write("Hello: " + TURKISH_TEXT, "ISO-8859-9");
```

Comment:
> Are we sure this charset is available on all environments? Maybe use a more neutral charset, such as "UTF-16", if you want something different from the default (which since Java 18 is "UTF-8" across all platforms)?

Both "ISO-8859-9" and "KOI8-R" should be supported by all Java implementations per https://docs.oracle.com/javase/8/docs/technotes/guides/intl/encoding.doc.html.